### DISCUSSION TITLE
What is the default URL for the frontend in a built Tauri 2.0 application?

### DISCUSSION THREAD

**Discussion by zeller-lorenz** [tauri-apps/tauri#11091] [Q&A] [answered]
Hi everyone,

I'm working on a desktop app using Tauri 2.0 and Angular. In the development environment, my Angular frontend runs on localhost:4200, which is configured in the tauri.conf.json.

However, after building the application for production, it no longer works because http://localhost:4200 obviously isn't available anymore.

My question is: What is the default URL or URI scheme for the frontend in a built Tauri 2.0 application? I want to ensure that the app correctly serves the static files after the build and that an OAuth2-based authentication with a redirectUri will work as expected.

Here's my build environment:

```typescript
// This file can be replaced during build by using the `fileReplacements` array.
// `ng build --prod` replaces `environment.ts` with `environment.prod.ts`.
// The list of file replacements can be found in `angular.json`.

import { environmentDefault } from './default';

export const environment = {
    ...environmentDefault,
    production: false,
    environmentName: 'TAURI_PROD',
    apiUrl: 'https://mydefaulturl.com',
    pngPrefixUrl: 'https://mydefaultpnguri.com/media/',
    identityServer: {
        issuer: 'https://auth.mydefaulturl.com/',
        redirectUri: 'http://localhost:4200/authentication-callback',
        postLogoutRedirectUri: 'http://localhost:4200',
        loginUrl: 'https://auth.mydefaulturl.com/authorize',
        clientId: 'dummyId',
        dummyClientSecret: 'dummyPass',
        responseType: 'code',
        oidc: false,
        scope: 'default',
        //autoSaveToken: true,
        tokenEndpoint: 'https://auth.mydefaulturl.com/token'
    }
};

/*
 * For easier debugging in development mode, you can import the following file
 * to ignore zone related error stack frames such as `zone.run`, `zoneDelegate.invokeTask`.
 *
 * This import should be commented out in production mode because it will have a negative impact
 * on performance if an error is thrown.
 */
import 'zone.js/plugins/zone-error'; // Included with Angular CLI.

```

The problem is probably the redirectUri `http://localhost:4200/authentication-callback`

Thanks for your help!

**Comment by FabianLars** [comment, marked as answer]
On windows and android it's `http://tauri.localhost/` and on linux/macos/ios it's `tauri://localhost/`

**Reply by zeller-lorenz** [reply]
Thank you!

**Reply by bm777** [reply]
thanks

**Reply by Haprog** [reply]
I was wondering why I can't find any clear information about this in the documentation. Just happened to stumble upon this difference in our project where I work on macOS and a colleague works on Windows and we noticed the URL is different.

I'd also like to know how does the `http(s)://tauri.localhost/` actually work within Tauri but you still can't access it from outside Tauri for example from a normal browser? Afaik `*.localhost` domains just point to localhost (`127.0.0.1` or `0.0.0.0`). I'd just like to know more about how this is implemented in Windows specifically to understand the security implications better and to know how to work with it in more advanced scenarios.

**Reply by FabianLars** [reply]
we're using webview2's `WebResourceRequested` event to catch those request before they hit the network stack. Once we drop Windows 7 support we will likely switch to `tauri://` schemes on windows too (probably with an option to switch back) to match the other platforms (except android which will still be stuck on http)
